src/Evolution/Systems/GrMhd/ValenciaDivClean/BoundaryConditions/FreeOutflow.cpp

```diff
@@ -0,0 +1,150 @@
+
+std::optional<std::string> FreeOutflow::dg_ghost(
+    const gsl::not_null<Scalar<DataVector>*> tilde_d,
+    const gsl::not_null<Scalar<DataVector>*> tilde_tau,
+    const gsl::not_null<tnsr::i<DataVector, 3, Frame::Inertial>*> tilde_s,
+    const gsl::not_null<tnsr::I<DataVector, 3, Frame::Inertial>*> tilde_b,
+    const gsl::not_null<Scalar<DataVector>*> tilde_phi,
+
+    const gsl::not_null<tnsr::I<DataVector, 3, Frame::Inertial>*> tilde_d_flux,
+    const gsl::not_null<tnsr::I<DataVector, 3, Frame::Inertial>*>
+        tilde_tau_flux,
+    const gsl::not_null<tnsr::Ij<DataVector, 3, Frame::Inertial>*> tilde_s_flux,
+    const gsl::not_null<tnsr::IJ<DataVector, 3, Frame::Inertial>*> tilde_b_flux,
+    const gsl::not_null<tnsr::I<DataVector, 3, Frame::Inertial>*>
+        tilde_phi_flux,
+
+    const gsl::not_null<Scalar<DataVector>*> lapse,
+    const gsl::not_null<tnsr::I<DataVector, 3, Frame::Inertial>*> shift,
+    const gsl::not_null<tnsr::II<DataVector, 3, Frame::Inertial>*>
+        inv_spatial_metric,
+
+    const std::optional<tnsr::I<DataVector, 3, Frame::Inertial>>&
+    /*face_mesh_velocity*/,
+    const tnsr::i<DataVector, 3, Frame::Inertial>&
+        outward_directed_normal_covector,
+    const tnsr::I<DataVector, 3, Frame::Inertial>&
+        outward_directed_normal_vector,
+
+    const Scalar<DataVector>& interior_rest_mass_density,
+    const Scalar<DataVector>& interior_specific_internal_energy,
+    const tnsr::I<DataVector, 3, Frame::Inertial>& interior_spatial_velocity,
+    const tnsr::I<DataVector, 3, Frame::Inertial>& interior_magnetic_field,
+    const Scalar<DataVector>& interior_lorentz_factor,
+    const Scalar<DataVector>& interior_pressure,
+    const Scalar<DataVector>& interior_specific_enthalpy,
+
+    const tnsr::I<DataVector, 3, Frame::Inertial>& interior_shift,
+    const Scalar<DataVector>& interior_lapse,
+    const tnsr::II<DataVector, 3, Frame::Inertial>&
+        interior_inv_spatial_metric) {
+  const size_t vector_size = get(interior_rest_mass_density).size();
+
+  // temp buffer to store
+  //  * n_i v^i
+  //  * v_{ghost}^i
+  //  * spatial metric
+  //  * sqrt of the determinant of spatial metric
+  //  * divergence cleaning field on ghost zone
+  Variables<tmpl::list<
+      ::Tags::TempScalar<0>, hydro::Tags::SpatialVelocity<DataVector, 3>,
+      gr::Tags::SpatialMetric<3>, ::Tags::TempScalar<1>, ::Tags::TempScalar<2>>>
+      temp_buffer{vector_size};
+  auto& normal_dot_interior_spatial_velocity =
+      get<::Tags::TempScalar<0>>(temp_buffer);
+  auto& exterior_spatial_velocity =
+      get<hydro::Tags::SpatialVelocity<DataVector, 3>>(temp_buffer);
+  auto& spatial_metric = get<gr::Tags::SpatialMetric<3>>(temp_buffer);
+  auto& sqrt_det_spatial_metric = get<::Tags::TempScalar<1>>(temp_buffer);
+  auto& exterior_divergence_cleaning_field =
+      get<::Tags::TempScalar<2>>(temp_buffer);
+
+  determinant_and_inverse(make_not_null(&sqrt_det_spatial_metric),
```

Comment:
Is it not possible to get these two from the interior? If not, that's fine, just curious :)

Reply:
Turns out.. it's possible :joy: 

I pushed a fixup since this comes with nontrivial amount of lines to be changed.

Reply:
Turns out.. it can.

I pushed this change as a fixup since it comes with nontrivial change of codes